sqlmesh/core/_typing.py

```diff
@@ -9,3 +10,8 @@
     SchemaName = t.Union[str, exp.Table]
     SessionProperties = t.Dict[str, t.Union[exp.Expression, str, int, float, bool]]
     CustomMaterializationProperties = t.Dict[str, t.Union[exp.Expression, str, int, float, bool]]
+
+if sys.version_info >= (3, 11):
+    from typing import Self as Self
```

Comment:
[Nit] Why do we need `as Self`?

Reply:
Because it gets removed on formatting for some reason. May be able to use `# noqa`, too, didn't try it.